src/core/border_router/routing_manager.hpp

```diff
@@ -538,6 +540,12 @@ class RoutingManager : public InstanceLocator
      *
      */
     Dhcp6PdState GetDhcp6PdState(void) const { return mPdPrefixManager.GetState(); }
+
+    void SetRequestDhcp6PdCallback(otBorderRoutingRequestDhcp6PdCallback aCallback)
```

Comment:
Better to avoid the C type definitions starting `otBorderRouting...` within `cpp` modules.

Let's add a `typedef` at the start of `RoutingManager` class definition close to other `typedef` definitions like this:

```c++
    typedef otBorderRoutingRequestDhcp6PdCallback Dhcp6PdCallback;  ///< DHCPv6 PD callback.
```

We can then use the `Dhcp6PdCallback` type within `RoutingManager` instead of the longer  `otBorderRoutingRequestDhcp6PdCallback` name (all the code in this file is already under namespace `ot::BorderRouter::`)